blog/content/post/2021-02-21-tigerlake-osf.md

```diff
@@ -0,0 +1,218 @@
+supposed to support GCC5 and newer versions, it occurred it is not always true.
+The freshly published code was not buildable with GCC when trying with the
+3mdeb [edk2-docker](https://github.com/3mdeb/edk2-docker). When giving it a
+little thought it is not surprising. Visual Studio compiler is the one that
+dominates the ecosystem of firmware. Intel, AMD and IBVs (Independent BIOS
+Vendors) use Windows and Microsoft compilers to build EDK2. Thus I think it was
+not tested with GCC compilers when published. The list of encountered problems
+is posted on [TianoCore Bugzilla](https://bugzilla.tianocore.org/show_bug.cgi?id=3220).
+
+Thankfully Intel engineers were very helpful and responsive on these bugs. The
+fixes for GCC toolchain were committed quickly (in just one week) to the Tiger
+Lake open board packages on edk2-platforms repository.
+
+In the mean time (before the fixes landed into repositories) I have fixed all
+compilation issues for GCC locally, I noticed that the packages contain support
+for LPDDR4 platform, again... The EDK2 FSP integration can work in two modes:
+
+* FSP API mode - the bootloader simply calls the entry point of the FSP module
+  by parsing the FSP header. The bootloader is also responsible for providing a
+  pointer to UPD values that have to be patched per platform.
+* FSP dispatch mode - this mode is dedicated for UEFI compliant bootloader
+  because the FSP acts as a standard Firmware Volume detectable by PEI
+  dispatcher. PEI dispatcher detects the FSP modules and evaluates the
+  dependency expressions of particular PE files and decides about the order of
+  their execution. The bootloader's responsibility is to provide necessary
+  protocol interfaces called Policy Updates that will set/patch the FSP UPD
+  values per platform needs.
+
+Switching between them is simply a build flag change. Gave a try to both, but
+unfortunately without success. The `FSP memory init returned error`, again...
+Changing the DDR memory signals routing as in coreboot did not help. Currently
+I am stuck at this problem which I have reported to the [Bugzilla](https://bugzilla.tianocore.org/show_bug.cgi?id=3219).
+Apparently the A0 stepping (which is an engineering sample) cannot work with
+the published code. It seems I have reached a dead end.
+
+## Possible solutions
+
+One of the obvious solutions that would be necessary is to add the build
+targets for TigerLake UP3 RVP DDR4 by Intel. To get rid of build issue under
+different toolchains a simple CI/CD would be more than enough to build test the
+reference platform with supported toolchains. I believe EDK2 has CI integration
+which could be extended to cover edk2-platforms repository. coreboot already
+has the build testing of the patchsets sent for review with Jenkins for a long
+time. It could be also great if Intel would publish validation results for FSP
+and open board packages. As far as I know, Intel publishes only the memory HCL
```

Comment:
Dasharo does it as part of its transparent validation philiosophy

Reply:
Added